index.js

```diff
@@ -34,11 +34,16 @@ exports.Socket = class TCPSocket extends Duplex {
 
     this._buffer = Buffer.alloc(readBufferSize)
 
+    this._multipleConnect = false
+    this._multipleConnectRemainingAttempts = []
+    this._multipleConnectErrors = []
```

Comment:
```suggestion
    this._addresses = null
    this._errors = null
```

This is sufficient; when multiple candidates exist, `_addresses` will be non-`null` and the next candidate will be the head of the list.